pkg/analyze/kube_resource.go

```diff
@@ -0,0 +1,140 @@
+package analyzer
+
+import (
+	"fmt"
+	"path/filepath"
+	"reflect"
+	"strconv"
+
+	"github.com/pkg/errors"
+	troubleshootv1beta2 "github.com/replicatedhq/troubleshoot/pkg/apis/troubleshoot/v1beta2"
+	iutils "github.com/replicatedhq/troubleshoot/pkg/interfaceutils"
+	"gopkg.in/yaml.v2"
+)
+
+func analyzeResource(analyzer *troubleshootv1beta2.ClusterResource, getFileContents func(string) (map[string][]byte, error)) (*AnalyzeResult, error) {
+
+	filemap := make(map[string]string)
+
+	filemap["Deployment"] = "deployments"
+	filemap["StatefulSet"] = "statefulsets"
+	filemap["NetworkPolicy"] = "network-policy"
+	filemap["Pod"] = "pods"
+	filemap["Ingress"] = "ingress"
+	filemap["Service"] = "services"
+	filemap["ResourceQuota"] = "resource-quotas"
+	filemap["Job"] = "jobs"
+	filemap["PersistentVoumeClaim"] = "pvcs"
+	filemap["pvc"] = "pvcs"
+	filemap["ReplicaSet"] = "replicasets"
+
+	targetFolder, ok := filemap[analyzer.Kind]
+
+	if ok != true {
+		return nil, errors.New("failed to find resource")
+	}
+
+	files, err := getFileContents(filepath.Join("cluster-resources", targetFolder, fmt.Sprintf("%s.json", analyzer.Namespace)))
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to read collected deployments from namespace")
+	}
+
+	var resource interface{}
+	for _, file := range files {
+		err = yaml.Unmarshal(file, &resource)
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to parse data as yaml doc")
+		}
+	}
+
+	var selected interface{}
+	items, _ := iutils.GetAtPath(resource, "items")
+	itemslice := items.([]interface{})
+	for _, item := range itemslice {
+		name, _ := iutils.GetAtPath(item, "metadata.name")
+		if name == analyzer.Name {
+			selected = item
+		}
+	}
+
```

Comment:
I think it would be handy to split the "get data from this resource" part of the analyzer to a separate function that other analyzers could make use of, and have this particular analyzer just the first example of using it for a simple object comparison.  That "get data from the resource" part that abstracts the file location away from the analyzer would be a good first step to making an API that can be called for use with sbctl, analyzers, etc.

Reply:
When splitting this out, would it be ok to leave it as a floating function (not associated with a specific analyzer) in t he analyze package? 

there's also considerations around parameters for this function. at the moment I'm having to pass through the GetFileContents function but I'd like to streamline the use of findResource() even if it means a slight duplication of code.